src/main/java/randoop/generation/DemandDrivenInputCreator.java

```diff
@@ -0,0 +1,406 @@
+  /**
+   * The principal set of sequences used to create other, larger sequences by the generator. New
+   * sequences are added on demand for creating object of missing types. Shared with {@link
+   * ComponentManager#gralComponents}.
+   */
+  private final SequenceCollection sequenceCollection;
+
+  /**
+   * A secondary sequence collection used to store sequences that are generated during the
+   * demand-driven input creation process. These sequences are added to the main sequence collection
+   * after the demand-driven input creation process is complete.
+   *
+   * <p>This is an optimization to reduce the search space for the missing types in the main
+   * sequence collection.
+   */
+  private final SequenceCollection secondarySequenceCollection;
+
+  /**
+   * If true, {@link #createInputForType(Type)} returns only sequences that declare values of the
+   * exact type that was requested.
+   */
+  private boolean exactTypeMatch;
+
+  /**
+   * If true, {@link #createInputForType(Type)} returns only sequences that are appropriate to use
+   * as a method call receiver, i.e., Type.isNonreceiverType() returns false for the type of the
+   * variable created by the sequence.
+   */
+  private boolean onlyReceivers;
+
+  /** Constructs a new {@code DemandDrivenInputCreator} object. */
+  public DemandDrivenInputCreator(
+      SequenceCollection sequenceCollection, boolean exactTypeMatch, boolean onlyReceivers) {
+    this.sequenceCollection = sequenceCollection;
+    this.secondarySequenceCollection = new SequenceCollection();
+    this.exactTypeMatch = exactTypeMatch;
+    this.onlyReceivers = onlyReceivers;
+  }
+
+  /**
+   * Performs a demand-driven approach for constructing input objects of a target type, when the
+   * sequence collection contains no objects of that type.
+   *
+   * <p>This method processes all available constructors and methods to identify possible ways to
+   * create objects of the {@code targetType}. For each method or constructor, it attempts to
+   * generate a sequence by searching for necessary inputs from the provided sequence collection,
+   * executing the sequence, and, if successful, storing it in the sequence collection for future
+   * use.
+   *
+   * <p>At the end of the process, it filters and returns the sequences that produce objects of the
+   * {@code targetType}, if any are found.
+   *
+   * <p>Here is the demand-driven algorithm in more detail:
+   *
+   * <ol>
+   *   <li>Initialize a worklist with the {@code targetType} and user-specified classes.
+   *   <li>Process types in the worklist:
+   *       <ul>
+   *         <li>Remove the next type. Skip this type if already processed or if it is a
+   *             non-receiver type.
+   *         <li>Identify constructors and methods that can produce objects of the current type or
+   *             the target type.
+   *         <li>Add input parameter types of these producer methods to the worklist.
```

Comment:
I think that adding to the beginning of the worklist would enable creation of more inputs.